cscs-checks/mch/alltoallv.py

```diff
@@ -2,92 +2,82 @@
 import reframe.utility.sanity as sn
 
 
-@rfm.parameterized_test([''], ['--nocomm'], ['--nocomp'])
+@rfm.parameterized_test(['default'], ['nocomm'], ['nocomp'])
 class Alltoallv(rfm.RegressionTest):
-    def __init__(self, exec_parameter):
+    def __init__(self, variant):
         super().__init__()
-        self.valid_systems = ['daint:gpu', 'dom:gpu', 'kesch:cn']
+        self.valid_systems = ['dom:gpu', 'daint:gpu', 'kesch:cn']
         self.valid_prog_environs = ['PrgEnv-gnu']
-        if self.current_system.name in ['daint', 'dom']:
-            self.modules = ['craype-accel-nvidia60']
-            self._pgi_flags = ['-acc', '-ta=tesla:cc60', '-Mnorpath']
-        elif self.current_system.name in ['kesch']:
-            self.modules = ['craype-accel-nvidia35']
-            self._pgi_flags = ['-O2', '-ta=tesla,cc35,cuda8.0']
+        self.variables = {'G2G': '1'}
+        self.executable = 'build/src/comm_overlap_benchmark'
+        if variant != 'default':
+            self.executable_opts = ['--' + variant]
 
-        self.num_tasks = 144
-        self.num_gpus_per_node = 16
-        self.num_tasks_per_node = 16
-        self.num_tasks_per_socket = 8
-        self.executable = ('build/src/comm_overlap_benchmark '
-                           '%s' % exec_parameter)
         self.sourcesdir = 'https://github.com/cosunae/comm_overlap_bench'
         # Checkout to the branch alltoallv
         self.prebuild_cmd = ['git checkout alltoallv']
         self.build_system = 'CMake'
         self.build_system.builddir = 'build'
-        self.build_system.config_opts = ['-DMPI_VENDOR=mvapich2',
-                                         '-DCUDA_COMPUTE_CAPABILITY="sm_37"',
-                                         '-DCMAKE_BUILD_TYPE=Release',
+        self.build_system.config_opts = ['-DCMAKE_BUILD_TYPE=Release',
                                          '-DENABLE_MPI_TIMER=ON']
-        self.build_system.max_concurrency = 1
+
+        if self.current_system.name == 'kesch':
+            self.num_tasks = 144
+            self.num_gpus_per_node = 16
+            self.num_tasks_per_node = 16
+            self.num_tasks_per_socket = 8
+            self.modules = ['craype-accel-nvidia35', 'cmake']
+            self.variables['MV2_USE_CUDA'] = '1'
+            self.build_system.config_opts += ['-DMPI_VENDOR=mvapich2',
+                                              '-DCUDA_COMPUTE_CAPABILITY="sm_37"']
+            self.build_system.max_concurrency = 1
+        else:
+            self.num_tasks = 4
+            self.num_gpus_per_node = 1
+            self.num_tasks_per_node = 1
+            self.modules = ['craype-accel-nvidia60', 'CMake']
+            self.variables['MPICH_RDMA_ENABLED_CUDA'] = '1'
+            self.build_system.config_opts += ['-DMPI_VENDOR=mvapich2',
```

Comment:
This is not correct, I should adapt the CMake options for the test on Daint/Domm.